### PR TITLE
[WIP] Expand cluster with given capacity 

### PR DESCRIPTION
**Pull request by annagitel**
wasn't tested yet

### CODE REVIEW

File: ocs_ci/ocs/machine.py [hunk 1]
```diff
@@ -152,60 +212,64 @@ def get_replica_count(machine_set):
...
-    cmd = f'''
-patch storagecluster/{storagecluster_name} --type='json' -p='[{{"op": "replace",
-"path": "/spec/storageDeviceSets/0/dataPVCTemplate/spec/resources/requests/storage", "value":{capacity}}}]'
-            '''
-    ocp.exec_oc_cmd(cmd)
-    return True
+    replica_count = get_replica_count(machine_set)
+    try:
+        for timer in TimeoutSampler(
+                300, 100, get_ready_replica_count, machine_set=machine_set
```
Comment (ebenahar) [inline ocs_ci/ocs/machine.py]: please delete redundant indentation

File: ocs_ci/ocs/resources/storage_cluster.py [hunk 2]
```diff
@@ -23,3 +30,76 @@ def __init__(self, resource_name="", *args, **kwargs):
         super(StorageCluster, self).__init__(
             resource_name=resource_name, kind='StorageCluster', *args, **kwargs
         )
+
+
+def add_capacity(capacity_string):
+    """
+        Add storage capacity to the cluster
```
Comment (ebenahar) [inline ocs_ci/ocs/resources/storage_cluster.py]: please align text with docstring indentation

File: ocs_ci/ocs/resources/storage_cluster.py [hunk 3]
```diff
@@ -23,3 +30,76 @@ def __init__(self, resource_name="", *args, **kwargs):
...
+
+        Args:
+            capacity_string(int): Size of the storage to add
+        Returns:
+            True if capacity was added, False if not
+
+    """
+
+    ocp = OCP(namespace=defaults.ROOK_CLUSTER_NAMESPACE, kind=constants.STORAGECLUSTER)
+    osd_capacity = parse_size_to_int(get_storage_cluster()['spec']['storageDeviceSets']
```
Comment (ebenahar) [inline ocs_ci/ocs/resources/storage_cluster.py]: Change to:

```python
osd_capacity = parse_size_to_int(
    get_storage_cluster().get('spec').get('storageDeviceSets')[0]
    .get('dataPVCTemplate').get('spec').get('resources').get('resources').get('storage')
)
```

File: ocs_ci/ocs/resources/storage_cluster.py [hunk 4]
```diff
@@ -23,3 +30,76 @@ def __init__(self, resource_name="", *args, **kwargs):
...
+                                        [0]['dataPVCTemplate']['spec']
+                                        ['resources']['resources']['storage'])
+
+    osd_replicas = parse_size_to_int(get_storage_cluster()['spec']['storageDeviceSets'][0]['replicas'])
```
Comment (ebenahar) [inline ocs_ci/ocs/resources/storage_cluster.py]: same

File: ocs_ci/ocs/resources/storage_cluster.py [hunk 5]
```diff
@@ -23,3 +30,76 @@ def __init__(self, resource_name="", *args, **kwargs):
...
+        else:
+            log.info("not enough worker nodes")
+            return False
+    else:
+        log.info("invalid storage capacity ")
+        return False
+
+
+def get_storage_cluster(namespace=defaults.ROOK_CLUSTER_NAMESPACE):
+    """
```
Comment (ebenahar) [inline ocs_ci/ocs/resources/storage_cluster.py]: align text with docstring

File: ocs_ci/ocs/resources/storage_cluster.py [hunk 6]
```diff
@@ -23,3 +30,76 @@ def __init__(self, resource_name="", *args, **kwargs):
...
+        Returns:
+            yaml: Storage cluster yaml
+    """
+
+    sc_obj = OCP(kind=constants.STORAGECLUSTER, namespace=namespace)
+    return sc_obj.get()
+
+
+def parse_size_to_int(num):
+    """
```
Comment (ebenahar) [inline ocs_ci/ocs/resources/storage_cluster.py]: same

File: ocs_ci/ocs/resources/storage_cluster.py (hunk 3 above)
Comment (annagitel) [reply ocs_ci/ocs/resources/storage_cluster.py]: all section was removed

File: ocs_ci/ocs/resources/storage_cluster.py (hunk 4 above)
Comment (annagitel) [reply ocs_ci/ocs/resources/storage_cluster.py]: all section was removed

**Pull request closed without merging**